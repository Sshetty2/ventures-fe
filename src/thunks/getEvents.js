import { setEvents, setLoading, setError } from '../actions';
import { fetchData } from '../utils/api';

export const getEvents = (event_type = '', month = '', state = '') => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      let path = (event_type.length || state.length || month.length) ? '?' : '';
      if (event_type.length) path = path + `&event_type=${event_type}`;
      if (month.length) path = path + `&month=${month}`;
      if (state.length) path = path + `&state=${state}`;
      const events = await fetchData('/events' + path, 'GET');
      dispatch(setEvents(events.data));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false))
  }
}