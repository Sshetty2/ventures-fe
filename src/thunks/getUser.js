import { setLoading, setError, setUser } from '../actions';
import { fetchData } from '../utils/api';

export const getUser = (userId = '') => { 
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const user = await fetchData(`/users/${userId}`, 'GET');
      dispatch(setUser(user.data.attributes));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
}