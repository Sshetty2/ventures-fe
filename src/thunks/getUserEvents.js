import { setLoading, setError, setUserEvents } from '../actions';
import { fetchData } from '../utils/api';

export const getUserEvents = (userId) => { 
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const userEvents = await fetchData(`/users/${userId}/events`, 'GET');
      dispatch(setUserEvents(userEvents.data));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
}