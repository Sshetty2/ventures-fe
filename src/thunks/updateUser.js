import { setLoading, setError, setUser } from '../actions';
import { fetchData } from '../utils/api';

export const updateUser = (user) => {
  return async (dispatch) => {
    const { name, bio } = user;
    try {
      dispatch(setLoading(true));
      const updatedUser = await fetchData(`/users/1?name=${name}&bio=${bio}`, 'PUT');
      dispatch(setUser(updatedUser.data.attributes));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
}