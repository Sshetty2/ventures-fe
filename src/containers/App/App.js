import React, { Fragment, useEffect } from 'react';
import SearchBar from '../SearchBar/SearchBar';
import Filters from '../Filters/Filters';
import EventContainer from '../EventContainer/EventContainer';
import Profile from '../Profile/Profile';
import EventPopUp from '../EventPopUp/EventPopUp';
import NavBar from '../../components/NavBar/NavBar';
import { withRouter, Route } from 'react-router-dom';
import { connect } from 'react-redux';
import { getEvents } from '../../thunks/getEvents';

export const App = ({ getEvents }) => {

  useEffect(() => {
    getEvents();
  })

  return (
    <div className="App">
      <NavBar />
      <Route exact path='/' render={() => 
        <Fragment>
          <SearchBar />
          <Filters />
          <EventContainer />
        </Fragment>
      } />
      <Route path='/profile' component={Profile} />
      <Route path='/event/:id' render={({ match }) => {
        const { id } = match.params;
        return <EventPopUp id={id} />
      }}
      />
    </div>
  );
}

const mapDispatchToProps = (dispatch) => ({
  getEvents: () => dispatch(getEvents())
});

export default withRouter(connect(null, mapDispatchToProps)(App));