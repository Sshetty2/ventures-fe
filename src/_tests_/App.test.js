import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from '../containers/App/App';
import { shallow } from 'enzyme';
import { getEvents } from '../thunks/getEvents';
jest.mock('../thunks/getEvents');

describe('App', () => {
  let wrapper;
  let mockError;
  let mockLocation;

  beforeEach(() => {
    mockError = '';
    mockLocation = { pathname: '/' };

    wrapper = shallow(
      <App
        getEvents={getEvents}
        error={mockError}
        location={mockLocation}
      />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when there is an error', () => {
    wrapper = shallow(<App error='Error fetching data.' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call window.location.reload when the Go Back button is clicked', () => {
    wrapper = shallow(<App error='Error fetching data.' />);
    window.location.reload = jest.fn();
    wrapper.find('.back-button').simulate('click');
    expect(window.location.reload).toHaveBeenCalledTimes(1);
  });

  describe('mapStateToProps', () => {
    it('should return an object of props', () => {
      const mockState = {
        error: 'Error message',
        otherState: false,
      };
      const expected = {
        error: 'Error message',
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when getEvents is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getEvents();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getEvents();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});