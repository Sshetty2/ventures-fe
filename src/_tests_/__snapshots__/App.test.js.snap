// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <NavBar />
  <Switch>
    <Route
      path="/profile"
      render={[Function]}
    />
    <Route
      path="/"
      render={[Function]}
    />
  </Switch>
  <Switch>
    <Route
      path="/login"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/event/:id"
    />
    <Route
      component={[Function]}
      path="/profile/upcoming/event/:id"
    />
    <Route
      component={[Function]}
      path="/profile/wishlist/event/:id"
    />
    <Route
      component={[Function]}
      path="/profile/past/event/:id"
    />
  </Switch>
</div>
`;

exports[`App should match the snapshot when there is an error 1`] = `
<div
  className="App"
>
  <NavBar />
  <div
    className="error-message"
  >
    <h1>
      Sorry, there was an error. Please try again.
    </h1>
    <button
      className="back-button"
      onClick={[Function]}
    >
      Go Back
    </button>
  </div>
</div>
`;
